### PR TITLE
Add nextflow access to data-bucket

### PR DESCRIPTION
**Pull request by alexiswl**
Give nextflow roles rw access to data bucket

### CODE REVIEW

**Comment by alexiswl** [comment]
@scwatts please shout out if you think these roles will need a greater scope

File: terraform/stacks/unimelb/data_archive/byob_ica_v2.tf [hunk 1]
```diff
@@ -616,6 +616,36 @@ data "aws_iam_policy_document" "development_data" {
...
+      // arn:aws:iam::843407916570:role/nextflow/oncoanalyser-pipeline-batch-instance-role
+      identifiers = [
+        "arn:aws:iam::${local.account_id_dev}:role/nextflow-oncoanalyser-pipeline-batch-instance-role",
+        "arn:aws:iam::${local.account_id_dev}:role/nextflow-oncoanalyser-task-batch-instance-role",
+        "arn:aws:iam::${local.account_id_dev}:role/nextflow-sash-pipeline-batch-instance-role",
+        "arn:aws:iam::${local.account_id_dev}:role/nextflow-sash-task-batch-instance-role",
+      ]
+    }
+    actions = [
+      "s3:GetBucketLocation",
```
Comment (reisingerf) [inline terraform/stacks/unimelb/data_archive/byob_ica_v2.tf:632]: This is needed?

File: terraform/stacks/unimelb/data_archive/byob_ica_v2.tf [hunk 2]
```diff
@@ -616,6 +616,36 @@ data "aws_iam_policy_document" "development_data" {
...
+      "s3:GetObject",
```
Comment (reisingerf) [inline terraform/stacks/unimelb/data_archive/byob_ica_v2.tf:633]: No `PutObject`?

Comment (alexiswl) [reply terraform/stacks/unimelb/data_archive/byob_ica_v2.tf:633]: Ah sorry, copied from the filemanager policy

**Comment by alexiswl** [comment]
I don't think I can apply this with my level of permissions? Or do I need to do this from the uom account?

**Review by scwatts — APPROVED** [review]
The roles look okay at a glance for required rw actions on S3 but approving with the qualifier I am no AWS expert

File: terraform/stacks/unimelb/data_archive/byob_ica_v2.tf (hunk 2 above)
Comment (reisingerf) [reply terraform/stacks/unimelb/data_archive/byob_ica_v2.tf:633]: Wrong statement?

File: terraform/stacks/unimelb/data_archive/byob_ica_v2.tf (hunk 1 above)
Comment (reisingerf) [reply terraform/stacks/unimelb/data_archive/byob_ica_v2.tf:632]: portal_data_access changes?

**Comment by reisingerf** [comment]
>I don't think I can apply this with my level of permissions? Or do I need to do this from the uom account?
Would be deployed from the UoM account, but not sure if you have enough permissions.
Happy to deploy when ready!

**Comment by alexiswl** [comment]
> Happy to deploy when ready!

Yes please, I only have read only access to the data account

File: terraform/stacks/unimelb/data_archive/byob_ica_v2.tf (hunk 1 above)
Comment (alexiswl) [reply terraform/stacks/unimelb/data_archive/byob_ica_v2.tf:632]: > portal_data_access changes?

Just fixed a typo

Comment (reisingerf) [reply terraform/stacks/unimelb/data_archive/byob_ica_v2.tf:632]: I meant that the last commit changed the permissions of the portal_data_access where I think it should have updated the permissions of the nextflow_batch statement...

Comment (alexiswl) [reply terraform/stacks/unimelb/data_archive/byob_ica_v2.tf:632]: Oh right, sorry, let me have a look at that

Comment (alexiswl) [reply terraform/stacks/unimelb/data_archive/byob_ica_v2.tf:632]: Ah so it did, force-pushing

File: terraform/stacks/unimelb/data_archive/byob_ica_v2.tf (hunk 2 above)
Comment (alexiswl) [reply terraform/stacks/unimelb/data_archive/byob_ica_v2.tf:631]: Sorry, resolved now

**Review by reisingerf — APPROVED** [review]
LGTM

**Comment by alexiswl** [comment]
@reisingerf have you made this change? 

Getting a S3 PutObject Permission Denied error - 

Also, 

If I put a 'aws sts get-caller-identity' in the middle of the script I get the ARN attribute as

```
"arn:aws:sts::843407916570:assumed-role/nextflow-oncoanalyser-pipeline-batch-instance-role/b1274523f5d540419d7dcd061932c63b"
```

Not sure if there is a different between `assumed-role` and `role` when granting roles access

**Comment by alexiswl** [comment]
Ah wait I need to give the role s3 access from the nf stack side right?

**Comment by alexiswl** [comment]
Hmm, still getting a permission denied error for both running the workflow from the aws cache directory and uploading to the analysis directory

**Comment by reisingerf** [comment]
Uh, I was waiting for you to merge the PR...

**Comment by reisingerf** [comment]
Deployed now.

**Pull request merged**